src/ui/components/DataTable/index.tsx

```diff
@@ -382,59 +454,25 @@ const DataTable = ({
 	}
 
 	const colOrderFiltered = state.colOrder.filter(
-		({ hidden, colIndex }) => !hidden && cols[colIndex],
+		({ hidden, colIndex }) => !hidden && state.cols[colIndex],
 	);
 
 	return (
 		<>
-			<CustomizeColumns
-				cols={cols}
-				colOrder={state.colOrder}
-				hasSuperCols={!!superCols}
-				show={state.showSelectColumnsModal}
-				onHide={() => {
-					setStatePartial({
-						showSelectColumnsModal: false,
-					});
-				}}
-				onReset={() => {
-					const newOrder = cols.map((col, i) => ({
-						colIndex: i,
-					}));
-					setStatePartial({
-						colOrder: newOrder,
-					});
-					state.settingsCache.set("DataTableColOrder", newOrder);
-				}}
-				onSortEnd={({ oldIndex, newIndex }) => {
-					const newOrder = arrayMoveImmutable(
-						state.colOrder,
-						oldIndex,
-						newIndex,
-					);
-					setStatePartial({
-						colOrder: newOrder,
-					});
-					state.settingsCache.set("DataTableColOrder", newOrder);
-				}}
-				onToggleHidden={(i: number) => () => {
-					const newOrder = [...state.colOrder];
-					if (newOrder[i]) {
-						newOrder[i] = {
-							...newOrder[i],
-						};
-						if (newOrder[i].hidden) {
-							delete newOrder[i].hidden;
-						} else {
-							newOrder[i].hidden = true;
-						}
+			{config ? (
+				<CustomizeColumns
+					config={config}
+					show={state.showSelectColumnsModal}
+					onHide={() => {
 						setStatePartial({
-							colOrder: newOrder,
+							showSelectColumnsModal: false,
 						});
-						state.settingsCache.set("DataTableColOrder", newOrder);
-					}
-				}}
-			/>
+					}}
+					onSave={() => {
+						location.reload();
```

Comment:
Rather than reloading, something like... `await realtimeUpdate(["customizeTable"]);`, and then make sure that, for every function in `src/worker/views` that returns data used in a table, it returns the updated data when `updateEvents` includes `"customizeTable"`.

Reply:
Made this change, however I'm not confident that this wont cause an infinite loops somewhere.

I wasnt able to figure out how to use [getDerivedStateFromProps](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops) with functional components so instead i made an effect which updates `state.cols` (in DataTable) when `props.cols` changes:

```
useEffect(() => {
	if (enableCustomizeColumns) {
		setStatePartial({ cols });
		processedRows = processRows();
	}
}, [cols]);
```

Reply:
Nvm this does not work on tradingBlock

Reply:
fixed tradingBlock